FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRInstallations.h"
+#import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseCore/FIRComponent.h>
+#import <FirebaseCore/FIRComponentContainer.h>
+#import <FirebaseCore/FIRLibrary.h>
+#import <FirebaseCore/FIRLogger.h>
+#import <FirebaseCore/FirebaseCore.h>
+
+#import "FIRInstallationsVersion.h"
+
+@protocol FIRInstallationsInstanceProvider
```

Comment:
What is this protocol for?

Reply:
@chliangGoogle  The protocol is needed to use `FIRComponent` to maintain single `FIRInstallations` per `FIRApp`. The same approach is used by other SDKs like Auth, IID, etc.